src/mpid/ch4/netmod/ofi/ofi_types.h

```diff
@@ -397,6 +350,23 @@ typedef struct {
     char kvsname[MPIDI_KVSAPPSTRLEN];
     char pname[MPI_MAX_PROCESSOR_NAME];
     int port_name_tag_mask[MPIR_MAX_CONTEXT_MASK];
+
+    /* Capability settings */
+#ifdef MPIDI_OFI_ENABLE_RUNTIME_CHECKS
```

Comment:
That's more or less what we're doing in [ofi_init.h](https://github.com/pmodels/mpich/pull/2374/files#diff-203a1d05f344fe7aef41d0a2ef017539R268). If the user selected a capability set/provider, the hints get filled in with the desired capabilities (pretty much accomplishing what you are talking about with the assertion) and passed in to OFI. If the user doesn't pick a capability set, those hints get ignored and we detect the capabilities with the result for initializing OFI.

For this particular line that you've put the comment on, we still need to account for whether or not we have runtime sets on because otherwise we don't need this part of the data structure. As we discussed on the phone, we could potentially remove this check since it's really only saving a few bytes. However, we still need the one above it to account for the potential for scalable endpoints.